packages/lisk-transactions/src/utils/sign_and_verify.ts

```diff
@@ -36,8 +45,109 @@ export const multiSignTransaction = (
 	return cryptography.signData(transactionHash, passphrase);
 };
 
+interface VerifySignatureReturn {
+	readonly verified: boolean;
+	readonly error?: TransactionError;
+}
+
+export const verifySignature = (
+	publicKey: string,
+	signature: string,
+	transactionBytes: Buffer,
+	id?: string,
+): VerifySignatureReturn => {
+	const transactionHash = cryptography.hash(transactionBytes);
+
+	const verified = cryptography.verifyData(
+		transactionHash,
+		signature,
+		publicKey,
+	);
+
+	return {
+		verified,
+		error: !verified
+			? new TransactionError(
+					`Failed to verify signature ${signature}`,
+					id,
+					'.signature',
+			  )
+			: undefined,
+	};
+};
+
+interface VerifyMultisignatureReturn {
+	readonly verified: boolean;
+	readonly errors: ReadonlyArray<TransactionError>;
+}
+
+export const verifyMultisignatures = (
+	memberPublicKeys: ReadonlyArray<string> = [],
+	signatures: ReadonlyArray<string>,
+	minimumValidations: number,
+	transactionBytes: Buffer,
+	id?: string,
+): VerifyMultisignatureReturn => {
+	if (!minimumValidations || typeof minimumValidations !== 'number') {
+		return {
+			verified: false,
+			errors: [new TransactionError('Sender does not have valid multimin', id)],
+		};
+	}
+	if (!signatures || (signatures && signatures.length < minimumValidations)) {
+		return {
+			verified: false,
+			errors: [new TransactionError('Missing signatures', id, '.signatures')],
+		};
+	}
+
+	const verifiedSignatures: string[] = [];
+	const checkedPublicKeys: string[] = [];
+
+	memberPublicKeys.forEach(publicKey => {
+		if (checkedPublicKeys.includes(publicKey)) {
+			return;
+		}
+
+		signatures.forEach((signature: string) => {
+			if (verifiedSignatures.includes(signature)) {
+				return;
+			}
+			const { verified: signatureVerified } = verifySignature(
+				publicKey,
+				signature,
+				transactionBytes,
+			);
+			if (signatureVerified) {
+				checkedPublicKeys.push(publicKey);
+				verifiedSignatures.push(signature);
+			}
+		});
+	});
+
+	const unverifiedSignatures = signatures.filter(
+		e => !verifiedSignatures.find(a => e === a),
+	);
+
+	return {
+		verified: verifiedSignatures.length >= minimumValidations,
```

Comment:
Could be that verified is true while this function still returns errors.

Reply:
in `base` it's checking the `errors` content as opposed to `verified` boolean, so it would fail with errors anyway. in some cases it could be helpful to know it passed multimin validations even if there were invalid signatures. what do you think?